internal/rootcoord/root_coord.go

```diff
@@ -2415,15 +2415,19 @@ func (c *Core) ReportImport(ctx context.Context, ir *rootcoordpb.ImportResult) (
 	}
 	colName = colMeta.GetSchema().GetName()
 
-	// When DataNode has done its thing, remove it from the busy node list.
-	func() {
-		c.importManager.busyNodesLock.Lock()
-		defer c.importManager.busyNodesLock.Unlock()
-		delete(c.importManager.busyNodes, ir.GetDatanodeId())
-		log.Info("dataNode is no longer busy",
-			zap.Int64("dataNode ID", ir.GetDatanodeId()),
-			zap.Int64("task ID", ir.GetTaskId()))
-	}()
+	// When DataNode has done its thing, remove it from the busy node list. And send import task again
+	if ir.GetState() == commonpb.ImportState_ImportPersisted {
```

Comment:
No need to check this. Import state is checked above (line #2595).

Also it the state check code above (around line #2595), please also remove busy node as well when state==Failed.